Geometry/HGCalCommonData/src/HGCalDDDConstants.cc

```diff
@@ -495,17 +496,25 @@ bool HGCalDDDConstants::isValidHex8(int layer, int modU, int modV, int cellU, in
 #endif
   if (!(jtr->second))
     return false;
+
   int N = ((hgpar_->waferTypeL_[itr->second] == 0) ? hgpar_->nCellsFine_ : hgpar_->nCellsCoarse_);
 #ifdef EDM_ML_DEBUG
   edm::LogVerbatim("HGCalGeom") << "HGCalDDDConstants::isValidHex8:Cell " << cellU << ":" << cellV << ":" << N
                                 << " Tests " << (cellU >= 0) << ":" << (cellU < 2 * N) << ":" << (cellV >= 0) << ":"
                                 << (cellV < 2 * N) << ":" << ((cellV - cellU) < N) << ":" << ((cellU - cellV) <= N);
 #endif
-  if ((cellU >= 0) && (cellU < 2 * N) && (cellV >= 0) && (cellV < 2 * N)) {
-    return (((cellV - cellU) < N) && ((cellU - cellV) <= N));
-  } else {
+  if ((cellU < 0) || (cellU >= 2 * N) || (cellV < 0) || (cellV >= 2 * N)) 
     return false;
-  }
+  if (((cellV - cellU) >= N) || ((cellU - cellV) > N))
+    return false;
+
+  auto ktr = hgpar_->waferTypes_.find(indx);
+  if (ktr == hgpar_->waferTypes_.end())
+    return true;
+
+  //  edm::LogVerbatim("HGCalGeom") << "Corners " << (ktr->second).first << ":" << waferVirtual(layer,modU,modV);
+  int type = ((itr == hgpar_->typesInLayers_.end()) ? 2 : hgpar_->waferTypeL_[itr->second]);
```

Comment:
Does `type == 2` have some meaning? Should 2 be a constant? It looks like `type` is boolean. It is checked for `type == 0` in `isValidCell8()`, so maybe its values should be 0 or 1, but not 2? Or will `type` later take on a wider range of values?

Reply:
Type could be 0, 1 or 2 for silicon wafers at least in this context. Type is one attribute of silicon cell.

Reply:
Is it worth having an `enum` for the three silicon cell types? Will silicon cell type be referenced elsewhere in the code?
If not, a short comment would be helpful here:
`// Silicon wafers can be type 0, 1, or 2.`

Reply:
I shall introduce enum later. Please sign this one. This PR solves an issue for HFNose detector. Need this one to be integrated fast

Reply:
Sounds good.